### PR TITLE
[stable/datadog] Remove default resources requests and limits

### PR DESCRIPTION
**Pull request by KIVagant**
#### What this PR does / why we need it:

Resources should not be limited by default because nobody knows where
the chart will be installed to. Especially worth not to set CPU limits
by default as per https://github.com/kubernetes/kubernetes/issues/67577

#### Special notes for your reviewer:

Null

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Review by hkaj — CHANGES_REQUESTED** [review]
Thanks for the PR @KIVagant ! 👍 for the CPU limits, that makes total sense. Could we keep the memory limits though? Unbounded memory usage makes me nervous. We don't have any memory leak but it happened in the past and I don't want the agent to cause trouble because of that.
Thanks again

**Comment by hkaj** [comment]
oh and you'll also need to signoff your commit to satisfy the PR lifecycle hooks :)

**Comment by KIVagant** [comment]
> Could we keep the memory limits though? Unbounded memory usage makes me nervous. 

Everyone should have the limits in any clusters but they shouldn't be predefined because someone can play with the chart even using local micro K8s. 

You can see how other charts follow the principle:

https://github.com/helm/charts/blob/master/stable/elasticsearch-curator/values.yaml#L79

> oh and you'll also need to signoff your commit to satisfy the PR lifecycle hooks :)

Hmm, I was sure I have done this but maybe not. I'll do on Monday.

**Comment by KIVagant** [comment]
DCO check finally passed

**Comment by cpanato** [comment]
@KIVagant can you please update your branch with the latest master and bump the chart version? thanks!

**Comment by hkaj** [comment]
Interesting, I didn't know about this recommendation.

We've seen users being surprised by high memory usage in some cases though (heavy tracing usage for example), and preventing the agent mem usage from growing unbounded is more important to us than making the deployment in minikube simpler.

Besides, it's still possible to override the limits at deployment time right? And IMHO the impact of forgetting to set limits in prod clusters is much worse than the impact of forgetting to unset limits in a minikube deployment.

Does that make sense to you?

**Comment by KIVagant** [comment]
> Does that make sense to you?

I can suggest the change, but it's always your choice as a maintainer. As for me, I prefer not to have the default limits at all anywhere. But at least, CPU limits should be removed until the related problem is fixed.

**Comment by KIVagant** [comment]
Rebased from master, bumped the chart version. Feel free to close the PR and if necessary create another one with partial changes.

**Review by cpanato — APPROVED** [review]

**Pull request merged**
